aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -94,6 +94,42 @@ void div_kernel(TensorIterator& iter) {
   }
 }
 
+void remainder_kernel(TensorIterator& iter) {
+  if (isIntegralType(iter.dtype(), /*includeBool*/ false)) {
+    AT_DISPATCH_INTEGRAL_TYPES(iter.dtype(), "remainder_cpu", [&]() {
+      cpu_kernel(iter, [](scalar_t a, scalar_t b) -> scalar_t {
+        TORCH_CHECK(b != 0, "ZeroDivisionError");
+        scalar_t r = a % b;
+        if ((a != 0) && (a < 0) != (b < 0)) {
+          r += b;
+        }
+        return r;
+      });
+    });
+  } else {
+    AT_DISPATCH_FLOATING_TYPES_AND_HALF(iter.dtype(), "remainder_cpu", [&]() {
+      cpu_kernel_vec(iter,
+        [=](scalar_t a, scalar_t b) __ubsan_ignore_float_divide_by_zero__ -> scalar_t {
+          if (b == 0) {
+            return std::numeric_limits<scalar_t>::quiet_NaN();
+          } else {
+            return a - b * at::native::floor_impl(a / b);
+          }
+        },
+        [=](Vec256<scalar_t> a, Vec256<scalar_t> b) {
+          Vec256<scalar_t> r = a - b * (a / b).floor();
+          Vec256<scalar_t> r_nans = b.map([](const scalar_t x) -> scalar_t {
+            return (x == static_cast<scalar_t>(0)) ?
+              std::numeric_limits<scalar_t>::quiet_NaN() :
+              static_cast<scalar_t>(0);
+          });
+          // Adding a NaN to any other number will result in NaN
+          return r + r_nans;
```

Comment:
The way I implemented the explicit divide by zero check for Vec256 might not be ideal. It would be better to be able to avoid all the 0 additions that this introduces. It would be nice if Vec256 had a binary input map function, so I could operate on pairs of elements in `a` and `b` directly. But I only see a unary map function available for Vec256.

Reply:
Oh never mind, I think I found what I need: `at::vec256::map2`.

Reply:
see my comment above, I'm not sure if this is necessary or not.